tests/CMakeLists.txt

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.13 FATAL_ERROR)
+
+project (az_iot_middleware_freertos_ut LANGUAGES C)
+
+enable_testing()
+
+set(AZ_IOT_REPO_ROOT "${PROJECT_SOURCE_DIR}/..")
+
+# GCC flags required via code coverage
+add_compile_options(-fprofile-arcs -ftest-coverage -O0 -g3)
+add_link_options(-fprofile-arcs -ftest-coverage)
+
+# FreeRTOS Kernel include
+include_directories(${AZ_IOT_REPO_ROOT}/libraries/FreeRTOS/FreeRTOS/Source/include)
+
+# port for FreeRTOS
+if(UNIX)
+    include_directories(${AZ_IOT_REPO_ROOT}/libraries/FreeRTOS/FreeRTOS/Source/portable/ThirdParty/GCC/Posix)
+else()
+    message( FATAL_ERROR "Tests only supported in linux." )
```

Comment:
Is the requirement making this Linux only the code coverage or are there other problems?  

UT should in theory be able to be run anywhere.  And I'm very attached to windbg :).

Reply:
Its more like cmake support that is required. More robust solution is what @danewalton is working on. For now to test PR and unit test test running, I am using this cmake.

---
In reply to: [607944775](https://github.com/Azure/azure-iot-middleware-freertos/pull/62#discussion_r607944775) [](ancestors = 607944775)